ngraph/core/reference/src/runtime/reference/convert.cpp

```diff
@@ -0,0 +1,273 @@
+                    jit_convert_f16_to_f32()
+                    {
+                        using namespace Xbyak;
+
+                        auto f16vec = xmm1;
+                        auto f32vec = ymm2;
+
+                        auto reg_src = rax;
+                        auto reg_dst = rbx;
+                        auto reg_sz = rdx;
+
+                        Label loop, tail, tail_loop, copy_loop, exit;
+
+                        preamble();
+
+                        mov(reg_src, ptr[param + offsetof(args_t, arg)]);
+                        mov(reg_dst, ptr[param + offsetof(args_t, out)]);
+                        mov(reg_sz, ptr[param + offsetof(args_t, count)]);
+
+                        L(loop);
+
+                        cmp(reg_sz, 8);
+                        jl(tail);
+
+                        movdqu(f16vec, xword[reg_src]);
+                        vcvtph2ps(f32vec, f16vec);
+                        vmovups(yword[reg_dst], f32vec);
+
+                        lea(reg_src, ptr[reg_src + sizeof(float16) * 8]);
+                        lea(reg_dst, ptr[reg_dst + sizeof(float) * 8]);
+
+                        sub(reg_sz, 8);
+                        jmp(loop);
+
+                        L(tail);
+
+                        test(reg_sz, reg_sz);
+                        jz(exit);
+
+                        sub(rsp, 8 * sizeof(float)); // allocate array for 8 floats on stack
+                        xor_(rsi, rsi);
+
+                        mov(qword[rsp], rsi);
+                        mov(qword[rsp + 8], rsi);
+                        mov(qword[rsp + 16], rsi);
+                        mov(qword[rsp + 24], rsi);
+
+                        // Tail conversion
+                        L(tail_loop);
+                        mov(di, word[reg_src + rsi * sizeof(float16)]); // read f16
+                        mov(word[rsp + rsi * sizeof(float16)], di);     // copy to stack
+                        inc(rsi);                                       // read next
+                        cmp(rsi, reg_sz);
+                        jl(tail_loop);
+
+                        movdqu(f16vec, xword[rsp]);
+                        vcvtph2ps(f32vec, f16vec);
+                        vmovups(yword[rsp], f32vec);
+
+                        xor_(rsi, rsi);
+
+                        // Tail copying
+                        L(copy_loop);
+                        mov(edi, dword[rsp + rsi * sizeof(float)]);
+                        mov(dword[reg_dst + rsi * sizeof(float)], edi);
+                        inc(rsi);
+                        cmp(rsi, reg_sz);
+                        jl(copy_loop);
+
+                        add(rsp, 8 * sizeof(float)); // Free the array on stack
+
+                        L(exit);
+
+                        postamble();
+                    }
+
+                public:
+                    typedef struct
+                    {
+                        const float16* arg;
+                        float* out;
```

Comment:
maybe better would be: `arg` -> `src`, `out` -> `dst`?

Reply:
fixed